L1Trigger/L1THGCal/src/be_algorithms/HGCalShowerShape.cc

```diff
@@ -207,268 +209,118 @@ float HGCalShowerShape::sigmaPhiPhiTot(const l1t::HGCalMulticluster& c3d) const
 
 float HGCalShowerShape::sigmaEtaEtaMax(const l1t::HGCalMulticluster& c3d) const {
 
-    const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
-
-    std::vector<int> layer ; // Size : ncl2D
-    std::vector<int> subdetID ;
-    std::vector<int> nTC ;
-    std::vector<float> tc_energy ; 
-    std::vector<float> tc_eta ;
-
-    for(const auto& clu : clustersPtrs){
-
-        layer.emplace_back(clu->layer());
-        subdetID.emplace_back(clu->subdetId());
+  std::map<int, vector<float> > tc_layer_energy;
```

Comment:
The ordering is not important here, so `std::unordered_map` can be used for better performance (hash table instead of binary tree)